File: file-handling/handling-text-files/python/read_text_file.py
# Raspberry Pi Foundation
# Developed to be used alongside Isaac Computer Science,
# part of the National Centre for Computing Education
# Usage licensed under CC BY-SA 4
     

def read_file():
    """Read and display all records from a file"""
    with open("playlist.txt", mode = "r") as playlist:
        for track in playlist:
            print(track)   
    

if __name__ == "__main__":
    read_file()

    

File: file-handling/handling-text-files/c-sharp/read_text_file.cs
/*
Isaac Computer Science
Usage licensed under the Open Government Licence v3.0

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;
using System.IO;

namespace MyApplication
{
    class Program
    {
        // The Main method is the entry point for all C# programs
        public static void Main() {
            ReadFile();
        }


        // Read and display each record from a file
        public static void ReadFile() {
            string filename = "playlist.txt";

            using (StreamReader sr = new StreamReader(filename)) {
                string line;
                while ((line = sr.ReadLine()) != null) {
                    Console.WriteLine(line);
                }
            } // The stream is now closed
        }

        
    }
}